control-plane/api-gateway/gatekeeper/gatekeeper_test.go

```diff
@@ -908,12 +939,90 @@ func TestUpsert(t *testing.T) {
 				roleBindings: []*rbac.RoleBinding{
 					configureRoleBinding(name, namespace, labels, "1"),
 				},
+				secrets:  []*corev1.Secret{},
 				services: []*corev1.Service{},
 				serviceAccounts: []*corev1.ServiceAccount{
 					configureServiceAccount(name, namespace, labels, "1"),
 				},
 			},
 		},
+		"create a new gateway with TLS certificate reference": {
```

Comment:
minor: can we add a test where the secret is in a separate namespace just to document how that works

Reply:
This is the backport of the main PR that already merged. Okay with adding this test in a separate PR so we can keep everything lined up?